File: Code/Arrays_Bit_Manipulation/Best_Time_To_Buy_And_Sell_Stock_I.java
import java.util.Scanner;

public class Main {

    public static int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for (int price : prices) {
            if (price < minPrice) {
                minPrice = price;
            } else if (price - minPrice > maxProfit) {
                maxProfit = price - minPrice;
            }
        }

        return maxProfit;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter number of days: ");
        int n = scanner.nextInt();
        int[] prices = new int[n];

        System.out.println("Enter stock prices:");
        for (int i = 0; i < n; i++) {
            prices[i] = scanner.nextInt();
        }

        int result = maxProfit(prices);
        System.out.println("Maximum profit: " + result);

        scanner.close();
    }
}

File: Code/Arrays_Bit_Manipulation/Best_Time_To_Buy_And_Sell_Stock_I.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = INT_MAX;
        int maxProfit = 0;

        for (int price : prices) {
            if (price < minPrice) {
                minPrice = price;
            } else if (price - minPrice > maxProfit) {
                maxProfit = price - minPrice;
            }
        }

        return maxProfit;
    }
};

int main() {
    Solution solution;
    vector<int> prices;
    int n, price;

    cout << "Enter number of days: ";
    cin >> n;

    cout << "Enter stock prices for each day:\n";
    for (int i = 0; i < n; ++i) {
        cin >> price;
        prices.push_back(price);
    }

    int result = solution.maxProfit(prices);
    cout << "Maximum profit: " << result << endl;

    return 0;
}
